src/Microsoft.AspNet.Html.Abstractions/HtmlContentBuilderExtensions.cs

```diff
@@ -132,5 +180,145 @@ private string DebuggerToString()
                 }
             }
         }
+
+        [DebuggerDisplay("{DebuggerToString()}")]
+        private class HtmlFormatString : IHtmlContent
+        {
+            private readonly IFormatProvider _formatProvider;
+            private readonly string _format;
+            private readonly object[] _args;
+
+            public HtmlFormatString(string format, object[] args)
+                : this(null, format, args)
+            {
+            }
+
+            public HtmlFormatString(IFormatProvider formatProvider, string format, object[] args)
+            {
+                if (format == null)
+                {
+                    throw new ArgumentNullException(nameof(format));
+                }
+
+                _formatProvider = formatProvider ?? CultureInfo.CurrentCulture;
+                _format = format;
+                _args = args;
+            }
+
+            public void WriteTo(TextWriter writer, IHtmlEncoder encoder)
+            {
+                if (writer == null)
+                {
+                    throw new ArgumentNullException(nameof(writer));
+                }
+
+                if (encoder == null)
+                {
+                    throw new ArgumentNullException(nameof(encoder));
+                }
+
+                var formatProvider = new EncodingFormatProvider(_formatProvider, encoder);
+                var builder = new StringBuilder();
+                builder.AppendFormat(formatProvider, _format, _args);
```

Comment:
`builder.AppendFormat` can't handle a `null` value and we don't `null` check it in the ctor.